db/migrate/20240122064113_create_ad_campaigns.rb

```diff
@@ -0,0 +1,35 @@
+class CreateAdCampaigns < ActiveRecord::Migration[7.1]
+  def change
+    create_table :ad_campaigns, :id => false, primary_key: :ad_campaign_id do |t| 
+      t.string        :ad_campaign_id, null: false
+      t.string        :ad_campaign_name
+      t.string        :ad_campaign_objective
+      t.datetime      :ad_campaign_startdate
```

Comment:
since I noticed `ad_campaign` as a prefix being used in all fields, given this is about ad_campaign (based on table name), the prefix in each field is redundant. Again not a problem for this assignment but something to keep in mind.

Reply:
Ack.